lib/connection.js

```diff
@@ -523,19 +522,15 @@ Connection.prototype._connect = function(address, sasl) {
   this.address = address;
   var self = this;
   self.dataHandler = self._receiveHeader;
-  var isSSL = address.protocol === 'amqps';
-  if (isSSL) {
-    var sslOptions = self._sslOptions || {};
-    sslOptions.port = self.address.port;
-    sslOptions.host = self.address.host;
-    self.client = tls.connect(sslOptions);
-    debug('Connecting to ' + self.address.host + ':' + self.address.port + ' via TLS');
-  } else {
-    self.client = net.connect({ port: self.address.port, host: self.address.host });
-    debug('Connecting to ' + self.address.host + ':' + self.address.port + ' via straight-up sockets');
+  self._transport = this._transportProvider.getTransportFor(address.protocol);
```

Comment:
since this was implemented as a singleton, you just need to do:
`TransportProvider.getTransportFor(address.protocol)`